File: Minerador/Paulo/events/event.py
import uuid
from datetime import datetime
from typing import Any, Dict, Optional

class Event:
    def __init__(self, event_type):
        self.id = str(uuid.uuid4())
        self.type = event_type
        self.payload: Dict[str, Any] = {}
        self.timestamp = datetime.now()
        self.result: Any = None
        self.success = True
        self.error_message: Optional[str] = None

    def with_payload(self, key: str, value: Any) -> 'Event':
        self.payload[key] = value
        return self

    def get(self, key: str) -> Any:
        return self.payload.get(key)

    def get_int(self, key: str) -> Optional[int]:
        v = self.payload.get(key)
        if v is None: return None
        try:
            return int(v)
        except (ValueError, TypeError):
            return None

    def get_double(self, key: str) -> Optional[float]:
        v = self.payload.get(key)
        if v is None: return None
        try:
            return float(v)
        except (ValueError, TypeError):
            return None

    def get_string(self, key: str) -> Optional[str]:
        v = self.payload.get(key)
        return str(v) if v is not None else None

    def get_boolean(self, key: str) -> Optional[bool]:
        v = self.payload.get(key)
        if isinstance(v, bool): return v
        if isinstance(v, str): return v.lower() == 'true'
        return None

    def set_result(self, result: Any):
        self.result = result

    def set_error(self, message: str):
        self.success = False
        self.error_message = message

    def __str__(self):
        return f"Event{{id={self.id}, type={self.type}, success={self.success}}}"

File: Paulo_Code/Prototipo_java_single-thread/src/main/java/br/pucminas/grafo/events/Event.java
package br.pucminas.grafo.events;

import java.time.Instant;
import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

/**
 * Envelope de evento que trafega pelo Event Bus.
 * Funciona como um "request/response" de API interna.
 */
public class Event {

    private final String id;
    private final EventType type;
    private final Map<String, Object> payload;
    private final Instant timestamp;

    // Campo de resposta preenchido pelo handler
    private Object result;
    private boolean success;
    private String errorMessage;

    public Event(EventType type) {
        this.id = UUID.randomUUID().toString();
        this.type = type;
        this.payload = new HashMap<>();
        this.timestamp = Instant.now();
        this.success = true;
    }

    // ── Builder fluente ────────────────────────────────────────────────────

    public Event with(String key, Object value) {
        payload.put(key, value);
        return this;
    }

    @SuppressWarnings("unchecked")
    public <T> T get(String key) {
        return (T) payload.get(key);
    }

    public Integer getInt(String key) {
        Object v = payload.get(key);
        if (v instanceof Integer i) return i;
        if (v instanceof String s)   return Integer.parseInt(s);
        return null;
    }

    public Double getDouble(String key) {
        Object v = payload.get(key);
        if (v instanceof Double d)  return d;
        if (v instanceof String s)  return Double.parseDouble(s);
        return null;
    }

    public String getString(String key) {
        Object v = payload.get(key);
        return v == null ? null : v.toString();
    }

    public Boolean getBoolean(String key) {
        Object v = payload.get(key);
        if (v instanceof Boolean b) return b;
        if (v instanceof String s)  return Boolean.parseBoolean(s);
        return null;
    }

    // ── Resposta ───────────────────────────────────────────────────────────

    public void setResult(Object result) {
        this.result = result;
    }

    public void setError(String message) {
        this.success = false;
        this.errorMessage = message;
    }

    @SuppressWarnings("unchecked")
    public <T> T getResult() {
        return (T) result;
    }

    // ── Getters ────────────────────────────────────────────────────────────

    public String getId()            { return id; }
    public EventType getType()       { return type; }
    public Instant getTimestamp()    { return timestamp; }
    public boolean isSuccess()       { return success; }
    public String getErrorMessage()  { return errorMessage; }
    public Map<String, Object> getPayload() { return payload; }

    @Override
    public String toString() {
        return "Event{id=" + id + ", type=" + type + ", success=" + success + "}";
    }
}
